lucene/core/src/java/org/apache/lucene/util/automaton/SortedIntSet.java

```diff
@@ -16,59 +16,70 @@
  */
 package org.apache.lucene.util.automaton;
 
+import java.util.Arrays;
 import java.util.TreeMap;
 import java.util.Map;
 import org.apache.lucene.util.ArrayUtil;
+import org.apache.lucene.util.SuppressForbidden;
 
 // Just holds a set of int[] states, plus a corresponding
-// int[] count per state.  Used by
-// BasicOperations.determinize
-final class SortedIntSet {
-  int[] values;
+// int[] count per state.  Used by Operations.determinize()
+final class SortedIntSet extends IntSet {
   int[] counts;
   int upto;
-  private int hashCode;
+
+  // Tracks if the hashCode computation is out of date and also if the array is out of sync with the map
```

Comment:
Isn't the array discarded as soon as we switch over to the map (so how could it be out of sync)?

Reply:
The array is used for equality comparison.